app/src/main/java/org/schabi/newpipe/local/dialog/PlaylistAppendDialog.java

```diff
@@ -179,10 +183,69 @@ private void onPlaylistSelected(@NonNull final LocalPlaylistManager manager,
                     .subscribe(ignored -> successToast.show()));
         }
 
-        playlistDisposables.add(manager.appendToPlaylist(playlist.uid, streams)
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(ignored -> successToast.show()));
+        /*Check if video had already added to this playlist*/
+        final StreamEntity videoToAdd = streams.get(0); //the first video is the target video to add
+        final long playListId = playlist.uid;
+        int duplicate = 0;
+
+        /*get all playlist videos*/
+        final List<PlaylistStreamEntry> entries = manager.getPlaylistStreams(playListId)
+                .firstElement()
+                .blockingGet();
+
+        /*count how many duplicate videos are there*/
+        for (final PlaylistStreamEntry entry : entries) {
```

Comment:
This would seem to get very slow if the playlist grows. Instead you should use a specific DB query which can make use of indexes&co